language/borrow-graph/src/graph/mod.rs

```diff
@@ -266,6 +192,58 @@ impl BorrowMap {
         }
     }
 
+    fn splice_out_intermediate(
+        &mut self,
+        parent_id: RefID,
+        parent_to_intermediate: &BorrowEdge<Loc, Lbl>,
+        child_id: RefID,
+        intermediate_to_child: &BorrowEdge<Loc, Lbl>,
+    ) {
+        let path = if parent_to_intermediate.strong {
+            paths::append(&parent_to_intermediate.path, &intermediate_to_child.path)
+        } else {
+            parent_to_intermediate.path.clone()
+        };
+        let strong = parent_to_intermediate.strong && intermediate_to_child.strong;
+        let loc = intermediate_to_child.loc;
+        let parent_to_child = BorrowEdge { strong, path, loc };
+        self.add_edge(parent_id, parent_to_child, child_id)
+    }
+
+    //**********************************************************************************************
+    // Subsumes/weakens
+    //**********************************************************************************************
+
+    /// checks if `self` covers `other`
+    pub fn leq(&self, other: &Self) -> bool {
+        self.unmatched_edges(other).is_empty()
+    }
+
+    fn unmatched_edges(&self, other: &Self) -> BTreeMap<RefID, BorrowEdges<Loc, Lbl>> {
+        let mut unmatched_edges = BTreeMap::new();
+        for (parent_id, other_ref) in &other.0 {
+            let self_ref = &self.0[parent_id];
+            let self_borrowed_by = &self_ref.borrowed_by.0;
+            for (child_id, other_edges) in &other_ref.borrowed_by.0 {
+                for other_edge in other_edges {
+                    let found_match = &self_borrowed_by[child_id]
+                        .iter()
+                        .any(|self_edge| self_edge.leq(other_edge));
+                    if !found_match {
+                        unmatched_edges
+                            .entry(*parent_id)
+                            .or_insert_with(BorrowEdges::new)
+                            .0
+                            .entry(*child_id)
+                            .or_insert_with(BTreeSet::new)
+                            .insert(other_edge.clone());
+                    }
+                }
+            }
+        }
+        unmatched_edges
+    }
```

Comment:
Anyone have an idea of writing this in a cleaner fashion?

It feels REAAALLLY close to a filter, but I couldn't quite get it to work